cpp/src_prims/sparse/csr.h

```diff
@@ -767,6 +773,9 @@ void weak_cc_label_batched(Type *labels, const Type *row_ind,
     //** swapping F1 and F2
     MLCommon::updateHost(host_fa, state->fa, N, stream);
     MLCommon::updateHost(host_xa, state->xa, N, stream);
+
+    CUDA_CHECK(cudaStreamSynchronize(stream));
```

Comment:
do we need this sync here?

Reply:
This one is a good question. I've been under the impression we always want to sync after we call `updateHost`. Is it safe to call `memCpyAsync` on the stream after `updateHost` is called?

Reply:
Typically you'd need a sync only after `updateHost` has been called *and* that buffer needs to be accessed from *host*. This doesn't seem to be the case here. So, I believe we can remove this sync.